app/models/log.rb

```diff
@@ -9,6 +9,7 @@ class Log < ApplicationRecord
   belongs_to :bulk_upload, optional: true
 
   before_save :update_status!
+  before_validation :verify_dsa_signed, on: :create
```

Comment:
can we use the full name in this method name for clarity to people who won't know what dsa means?